tests/test_crawl.py

```diff
@@ -226,3 +226,13 @@ def cb(response):
         s = dict(est[0])
         self.assertEqual(s['engine.spider.name'], crawler.spider.name)
         self.assertEqual(s['len(engine.scraper.slot.active)'], 1)
+
+    @defer.inlineCallbacks
+    def test_graceful_crawl_error_handling(self):
+        crawler = get_crawler(SimpleSpider)
+        class TestError(Exception):
+            pass
+        with mock.patch('scrapy.crawler.ExecutionEngine.open_spider') as mock_os:
```

Comment:
It simulates an exception being thrown somewhere inside the `try` block of `Crawler.crawl()`, after the engine has been created. It could be in `self.spider.start_requests()`, or (as mocked here) in `self.engine.open_spider()`. E.g. in the scheduler class's `from_crawler()`, in a spider middleware's `process_start_requests()`, in a pipeline's `open_spider()`, etc.

(I have trouble reading some mock tests too, but I liked [this tutorial](http://www.toptal.com/python/an-introduction-to-mocking-in-python) :))